runtime/common/src/auctions.rs

```diff
@@ -201,7 +201,7 @@ pub mod pallet {
 			// If the current auction was in its ending period last block, then ensure that the (sub-)range
 			// winner information is duplicated from the previous block in case no bids happened in the
 			// last block.
```

Comment:
Wouldn't it be in case no bids happen in this bock, not the last one?

Reply:
This is `on_initialize`, so when this function runs, there would be no transactions

Reply:
Yup, that totally makes sense. My hang up then is just on 

> in case no bids happened in the last block.

In the ending period don't we always want to initialize the wining data for the block based on the last block, regardless of if any bids happened? 

(btw, not really concerned with changing the comment, just want to make sure I understand the system :))

Reply:
The ending period has multiple samples and sub-samples (as you might see in some of the changes we made).

When entering a new sample, if there is not any data, we want to backfill it, but within a subsample, if there is already data, we want to update it.

I believe that is what the logic is doing here. We dont want to over-write any data, but we do want to propogate data when it does not exist.

Sample is not always 1:1 with block